data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java

```diff
@@ -59,29 +59,32 @@ public BeaconChainHeadHandler(ChainStorageClient client) {
   @Override
   public void handle(Context ctx) {
     Bytes32 head_block_root = client.getBestBlockRoot();
+    if (head_block_root == null) {
+      ctx.status(SC_NO_CONTENT);
+      return;
+    }
 
-    UnsignedLong head_block_slot = client.getBestSlot();
-    UnsignedLong finalized_epoch = client.getFinalizedEpoch();
-    Bytes32 finalized_root = client.getFinalizedRoot();
-    UnsignedLong justified_epoch = client.getJustifiedEpoch();
-    Bytes32 justified_root = client.getJustifiedRoot();
+    // derive all other state from the head_block_root
+    BeaconState beaconState = client.getStore().getBlockState(head_block_root);
+    Checkpoint finalizedCheckpoint = beaconState.getFinalized_checkpoint();
+    Checkpoint justifiedCheckpoint = beaconState.getCurrent_justified_checkpoint();
 
-    if (head_block_root == null || finalized_root == null || justified_root == null) {
+    if (finalizedCheckpoint.getRoot() == null || justifiedCheckpoint.getRoot() == null) {
```

Comment:
I'd have expected that `finalizedCheckpoint` and `justifiedCheckpoint` are `null` not just the `root` inside them. But if we already have a `BestBlockRoot` then we must be post-genesis and so not need any `null` checks here (at genesis the justified and finalised checkpoints are set to the genesis state and they move forward from there).

Reply:
removed that check